lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -241,6 +241,22 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     metric.putMetric(`${this.metricPrefix}_send_${method}`, 1, MetricLoggerUnit.Count)
   }
 
+  logRpcResponseMatch(method: string, otherProvider: SingleJsonRpcProvider) {
+    metric.putMetric(
+      `${this.metricPrefix}_other_provider_${otherProvider.providerId}_method_${method}_rpc_match`,
+      1,
+      MetricLoggerUnit.Count
+    )
+  }
+
+  logRpcResponseMismatch(method: string, otherProvider: SingleJsonRpcProvider) {
+    metric.putMetric(
+      `${this.metricPrefix}_other_provider_${otherProvider.providerId}_method_${method}_rpc_mismatch`,
+      1,
+      MetricLoggerUnit.Count
+    )
+  }
+
```

Comment:
should these 2 be in `UniJsonRpcProvider`? it kind of feels like `SingleJsonRpcProvider` doesn't need to know about these methods but I could be convinced otherwise

Reply:
Eventually all those stuffs should be migrated to the standalone RPC service. I'm not too worried about the small gymnastics like this.